lib/core/connector_job.rb

```diff
@@ -133,5 +148,24 @@ def self.fetch_jobs_by_query(query, page_size)
 
       results
     end
+
+    def initialize(es_response)
+      # TODO: remove the usage of with_indifferent_access. The initialize method should expect a hash argument
+      @elasticsearch_response = es_response.with_indifferent_access
+    end
+
+    def terminate!(status, error = nil, ingestion_stats = {}, connector_metadata = {})
+      ingestion_stats ||= {}
```

Comment:
This method is not tested and I believe it should be - it has some logic about defaulting `intestion_stats`, `total_document_count`, conditional `cancelled_at` and `metadata` assignments.

Reply:
Will add test soon